src/app/header/header.component.ts

```diff
@@ -36,7 +38,9 @@ export class HeaderComponent implements OnInit {
     public context: ContextService
   ) {
     router.events.subscribe((val) => {
-      this.onNavigate();
+      if (val instanceof NavigationEnd) {
```

Comment:
You can "type" the variable like "val:NavigationEnd".  I haven't tested it will be enough of a check. But is worth know you can do it.